tools/CheckDates.scala

```diff
@@ -0,0 +1,184 @@
+import sys.process._
+
+object CheckDates {
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change match regex "\\[*.\\]" (e.g. '[Tech message]')  it will be ignored and 
+    we will take next commit.
+
+    analyzeDates - just shows if anything is ok
+    fixWrongDates - actual fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyzeDates
+           scala tools/CheckDates.scala fixWrongDates
+
+  """
+  trait DateComparison {
+    def actualLastUpdateYear: String
+    def gitLastUpdateYear: String
+    def actualCreationYear: String
+    def gitCreationYear: String
+
+    def ifUpdateDateWrong(): Boolean = {
+      actualLastUpdateYear.toInt != gitLastUpdateYear.toInt
+    }
+
+    def ifUpdateDateWrongStr(): Boolean = {
+      actualLastUpdateYear != gitLastUpdateYear
+    }
+
+    def ifOneYear(): Boolean = {
+      gitCreationYear != gitLastUpdateYear
+    }
+  }
+
+  case class FileDates(pathToFile: String,
+                      actualLastUpdateYear: String,
+                      gitLastUpdateYear: String,
+                      actualCreationYear: String,
+                      gitCreationYear: String) extends DateComparison
+
+  case class GitInfo(message: String, date: String)
+
+  def findRootFolder(): String = {
```

Comment:
If there assumptions about where this command should be run from add a guide to the `usage` string. Otherwise consider adding a command option to avoid the root to be specified.

`CopyrightDateTool ~/workspace/scala analyze`